### PR TITLE
docs: add development guide for ui-components

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref #44668

This PR adds a development guide for `ui-components`. 

I think I will revisit the doc and make it more comprehensive as we go. We have been gradually standardizing the development process, but we are in an early phase still, so I'm keeping the doc high-level for now to avoid overwhelming new contributors.

I'm not a native English speaker, so please help check for grammar errors or what could be improved 🙂.

### CODE REVIEW

File: docs/how-to-work-on-the-component-library.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# How to Work on the Component Library
+
+freeCodeCamp has been using Bootstrap components in the UI. However, we are moving away from it and building our own component library, which helps standardize our UX/UI patterns and improve the accessibility. The project is tracked in [this GitHub issue](https://github.com/freeCodeCamp/freeCodeCamp/issues/44668).
```
Comment (raisedadead) [inline docs/how-to-work-on-the-component-library.md:3]:
```suggestion
> [!NOTE]
>
> freeCodeCamp has been using Bootstrap components in the UI. However, we are moving away from it and building our own component library, which helps standardize our UX/UI patterns and improve accessibility. The project is tracked in [this GitHub issue](https://github.com/freeCodeCamp/freeCodeCamp/issues/44668).
```

File: docs/how-to-work-on-the-component-library.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+The new library is called `ui-components`. The components are built mostly from scratch with basic HTML elements and [Tailwind CSS](https://tailwindcss.com/). The following steps are recommended when working on a new component:
```
Comment (raisedadead) [inline docs/how-to-work-on-the-component-library.md:5]:
```suggestion
The following steps are recommended when working on a new component:
```

File: docs/how-to-work-on-the-component-library.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# How to Work on the Component Library
```
Comment (raisedadead) [inline docs/how-to-work-on-the-component-library.md:1]:
```suggestion
Welcome to freeCodeCamp's `ui-components` library. The components are built mostly from scratch with basic HTML elements and [Tailwind CSS](https://tailwindcss.com/). 

# How to Work on the Component Library
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
Quick nit-picks.

**Review by raisedadead — DISMISSED** [review]

**Review by Ismailtlem — COMMENTED** [review]
One small thing

File: docs/how-to-work-on-the-component-library.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+- Support forwarding ref
+- Be styled for both light and dark themes
+- Be styled internally based on their props (the consumers should not need to restyle the components with the `className` prop)
+- Utilize the built-in styling system from Tailwind instead of having custom styles
+
+### Useful links
+
+- [Tailwind CSS Configuration](https://tailwindcss.com/docs/configuration)
+- [React Bootstrap v0.33 Docs](https://react-bootstrap-v3.netlify.app/components/)
```
Comment (Ismailtlem) [inline docs/how-to-work-on-the-component-library.md:56]: this url : https://react-bootstrap-v3.netlify.app/components/ doesn't exist, maybe just https://react-bootstrap-v3.netlify.app/ is enough

File: docs/how-to-work-on-the-component-library.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+- Support forwarding ref
+- Be styled for both light and dark themes
+- Be styled internally based on their props (the consumers should not need to restyle the components with the `className` prop)
+- Utilize the built-in styling system from Tailwind instead of having custom styles
+
+### Useful links
+
+- [Tailwind CSS Configuration](https://tailwindcss.com/docs/configuration)
+- [React Bootstrap v0.33 Docs](https://react-bootstrap-v3.netlify.app/)
```
Comment (huyenltnguyen) [inline docs/how-to-work-on-the-component-library.md:56]:
```suggestion
- [React Bootstrap v0.33 Docs](https://react-bootstrap-v3.netlify.app)
```

**Review by raisedadead — APPROVED** [review]

**Review by Ismailtlem — APPROVED** [review]
LGTM

**Pull request merged**